designer/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentsUsageHelper.scala

```diff
@@ -25,11 +27,43 @@ object ComponentsUsageHelper {
       }
     }
 
-    processesDetails
+    @tailrec
+    def mergeFragments(fragmentsProcessesDetails: List[BaseProcessDetails[ScenarioComponentsUsages]],
+                       acc: Map[ComponentId, List[(BaseProcessDetails[Unit], List[NodeId])]]): Map[ComponentId, List[(BaseProcessDetails[Unit], List[NodeId])]] =
+      fragmentsProcessesDetails match {
```

Comment:
What gives you this pattern matching? It won't give a big performance improvement and instead obfuscates the code a bit

Reply:
I wanted to avoid `if-else` statement and did list `head-tail` destruct. What you suggest instead?

Reply:
Sorry, I misunderstand it. It is related with my another question about `componentIdUsages` preparation